Initial build for data catalog script

meoflaxz opened the pull request.

izztsfn commented on Data_Catalog/data_catalog.sql:
```diff
@@ -0,0 +1,119 @@
+CREATE OR REPLACE TABLE `x-marketing.data_catalog.data_catalog` AS  
```
Use Insert statement instead of Create

izztsfn commented on Data_Catalog/data_catalog.sql:
```diff
@@ -0,0 +1,119 @@
+CREATE OR REPLACE TABLE `x-marketing.data_catalog.data_catalog` AS  
+WITH main_data AS (
+  WITH column_name AS (
```
Avoid using nested CTE

izztsfn commented on Data_Catalog/data_catalog.sql:
```diff
@@ -0,0 +1,119 @@
+    AND column_name.table = table_created_time.table_name
+),
+crm_combined AS (
+  WITH hubspot_source AS (
```
create a separated CTE to combine all of the CRMs to avoid subquerying

izztsfn reviewed (changes requested):
Just some formatting changes to make

meoflaxz commented:
As per latest commit: https://github.com/2x-data-ops/dataops/pull/213/commits/a38efa8af4211e0af3df7be94cd5bc34643d9648

Fixed:
1. Remove both nested CTE
2. Now use INSERT statement instead of CREATE OR REPLACE 

Added:
1. Changed column naming convention
2. Table now has partitioning (table_created) and clustering (_table, _client_name, _crm_source) specs

@izztsfn

izztsfn reviewed (approved):
LGTM. I learned a lot from this script also. Keep up the good work! @meoflaxz
